crates/wash-cli/src/call.rs

```diff
@@ -372,119 +363,120 @@ impl HttpHandlerInvocationOpts {
+    let result = tokio::time::timeout(
+       std::time::Duration::from_millis(timeout_ms),
+        client
+            .invoke_handle_http(Some(gen_wash_call_headers()), request)
+  )
+   .await
+   .with_context(|| format!("component invocation timeout, is component [{component_id}] running in lattice [{lattice}]?"))?
+   .context("failed to perform HTTP request")?;
+
+    match result {
+        (Ok(mut resp), _errs, io) => {
+            if let Some(io) = io {
+                io.await.context("failed to complete async I/O")?;
+            }
+
+            let status = resp.status().as_u16();
+            let headers =
+                HashMap::<String, String>::from_iter(resp.headers().into_iter().map(|(k, v)| {
+                    (
+                        k.as_str().into(),
+                        v.to_str().map(|v| v.to_string()).unwrap_or_default(),
+                    )
+                }));
+
+            // Read the incoming body into a string
+            let mut body = BytesMut::new();
+            while let Some(bytes) = resp.body_mut().body.next().await {
+                body.extend(bytes);
+            }
+            let body = body.freeze();
+
+            // If the option for parsing the response as JSON was provided, parse it directly,
+            // and return that as JSON
+            let output = if extract_json {
+                let body_json = serde_json::from_slice(&body)
+                    .context("failed to parse response body bytes into a valid JSON object")?;
+                CommandOutput::new(
+                    serde_json::to_string_pretty(&body_json)
+                        .context("failed to print http response JSON")?,
+                    HashMap::from([("response".into(), body_json)]),
+                )
+            } else {
+                let http_resp = HttpResponse {
+                    status,
+                    headers,
+                    body,
+                };
+                CommandOutput::new(
+                    serde_json::to_string(&http_resp)
+                        .context("failed to print http response JSON")?,
+                    HashMap::from([(
+                        "response".into(),
+                        serde_json::to_value(&http_resp)
+                            .context("failed to convert http response to value")?,
+                    )]),
+                )
+            };
+
+            Ok(output)
+        }
+        // For all other responses, something has gone wrong
+        _ => bail!("unexpected response after HTTP wRPC invocation"),
+    }
+}
+
+/// Invoke a wRPC endpoint that takes nothing and returns a string
+async fn wrpc_invoke_simple(
+    client: wrpc_transport_nats::Client,
+    lattice: &str,
+    component_id: &str,
+    instance: &str,
+    function_name: &str,
+    timeout_ms: u64,
+) -> Result<CommandOutput> {
+    let result = tokio::time::timeout(
+       std::time::Duration::from_millis(timeout_ms),
```

Comment:
Is there still a reason to wrap this in `tokio::time::timeout` if we're calling `timeout` on the `Invoke`?

Reply:
Nope, this could be removed!

- [ ] pass timeout directly to wrpc client